### PR TITLE
Feat: terminal connection integration

### PR DESCRIPTION
**Pull request by powerfooI**
## Summary

1. Introduced xterm to cluster and tenant connection.
2. Handled terminal creation and connection for cluster and tenant.

Wait for PR #264 to be merged

### CODE REVIEW

File: ui/src/pages/Cluster/Detail/Connection/index.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import React, { useState } from 'react'
+import { PageContainer } from '@ant-design/pro-components'
+import { intl } from '@/utils/intl'
+import { OBTerminal } from '@/components/Terminal/terminal'
+import { Button, Row, message } from 'antd'
+import { request } from '@umijs/max'
+import { useRequest } from 'ahooks'
+import { getClusterDetailReq } from '@/services'
+import BasicInfo from '../Overview/BasicInfo'
+import { useParams } from '@umijs/max'
```
Comment (yang1666204) [inline ui/src/pages/Cluster/Detail/Connection/index.tsx:10]: It should be imported together with request

File: ui/src/components/Terminal/terminal.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      const term = new Terminal({
+        cols: 160,
+        rows: 60,
+      })
+      term.open(ref.current)
+
+      if (term.options.fontSize) {
+        const containerWidth = ref.current.clientWidth
+
+        const cols = Math.floor(containerWidth / 9.2)
```
Comment (yang1666204) [inline ui/src/components/Terminal/terminal.tsx:33]: How are 9 points 2 and 4 calculated?

File: ui/src/pages/Cluster/Detail/Connection/index.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  const {ns, name} = useParams();
+
+  const { data: clusterDetail } = useRequest(getClusterDetailReq, {
+    defaultParams: [{ name: name!, ns: ns! }],
+  })
+
+  const { runAsync } = useRequest(async (): Promise<{
+    data: { terminalId: string }
+  }> => {
+    return request(`/api/v1/obclusters/namespace/${ns}/name/${name}/terminal`, {
```
Comment (yang1666204) [inline ui/src/pages/Cluster/Detail/Connection/index.tsx:31]: Is it better to put it in the services directory?

File: ui/src/components/Terminal/terminal.tsx (hunk 2 above)
Comment (powerfooI) [reply ui/src/components/Terminal/terminal.tsx:33]: I got the magic ratio through some experiments. `9.2` gave a good effect.

File: ui/src/pages/Cluster/Detail/Connection/index.tsx (hunk 3 above)
Comment (yang1666204) [reply ui/src/pages/Cluster/Detail/Connection/index.tsx:30]: ok

**Review by yang1666204 — APPROVED** [review]

**Pull request merged**